web-bundle/src/main/java/com/graphhopper/resources/MapboxResource.java

```diff
@@ -0,0 +1,181 @@
+        if (!bannerInstructions)
+            throw new IllegalArgumentException("You need to enable banner instructions right now");
+
+        double minPathPrecision = 1;
+        if (overview.equals("full"))
+            minPathPrecision = 0;
+
+        String vehicleStr = convertProfileToGraphHopperVehicleString(profile);
+        List<GHPoint> requestPoints = getPointsFromRequest(httpReq, profile);
+
+        StopWatch sw = new StopWatch().start();
+
+        // TODO: initialization with heading/bearings
+        // TODO: how should we use the "continue_straight" parameter? This is analog to pass_through, would require disabling CH
+
+        GHRequest request = new GHRequest(requestPoints);
+        request.setVehicle(vehicleStr).
+                setWeighting(weighting).
+                setLocale(localeStr).
+                getHints().
+                put(CALC_POINTS, true).
+                put(INSTRUCTIONS, enableInstructions).
+                put(WAY_POINT_MAX_DISTANCE, minPathPrecision);
+
+        GHResponse ghResponse = graphHopper.route(request);
+
+        float took = sw.stop().getSeconds();
+        String infoStr = httpReq.getRemoteAddr() + " " + httpReq.getLocale() + " " + httpReq.getHeader("User-Agent");
+        String logStr = httpReq.getQueryString() + " " + infoStr + " " + requestPoints + ", took:"
+                + took + ", " + weighting + ", " + vehicleStr;
+
+        if (ghResponse.hasErrors()) {
+            logger.error(logStr + ", errors:" + ghResponse.getErrors());
+            // Mapbox specifies 422 return type for input errors
+            return Response.status(422).entity(MapboxResponseConverter.convertFromGHResponseError(ghResponse)).
+                    header("X-GH-Took", "" + Math.round(took * 1000)).
+                    build();
+        } else {
+            return Response.ok(MapboxResponseConverter.convertFromGHResponse(ghResponse, translationMap, request.getLocale())).
+                    header("X-GH-Took", "" + Math.round(took * 1000)).
+                    build();
+        }
+    }
+
+    /**
+     * This method is parsing the request URL String. Unfortunately it seems that there is no better options right now.
+     * See: https://stackoverflow.com/q/51420380/1548788
+     *
+     * The url looks like: ".../{profile}/1.522438,42.504606;1.527209,42.504776;1.526113,42.505144;1.527218,42.50529?.."
+     */
+    private List<GHPoint> getPointsFromRequest(HttpServletRequest httpServletRequest, String profile) {
+
+        String url = httpServletRequest.getRequestURI();
+        url = url.replaceFirst("/mapbox/directions/v5/mapbox/" + profile + "/", "");
+        url = url.replaceAll("\\?[*]", "");
+
+        String[] pointStrings = url.split(";");
+
+        List<GHPoint> points = new ArrayList<>(pointStrings.length);
+        for (int i = 0; i < pointStrings.length; i++) {
+            points.add(GHPoint.fromStringLonLat(pointStrings[i]));
+        }
+
+        return points;
+    }
```

Comment:
Will the following work?

`@Path("/{profile}/{coordinatesString}")`

and then String[] pointStrings = coordinatesString.split(";"); 

This way at least the first three lines can be avoided.

Reply:
No, this was my first idea as well :).

The problem is that the path parameter coordinate string would only contain the first coordinate up to the first `;`, everything else is parsed as matrix parameter, which is unsorted.

Reply:
Oh, ugly. And this means `{coordinatesArray: .*}` is no longer in use?

Reply:
Yes, the parameter `{coordinatesArray: .*}` is no longer used. I could rename it or see if I can remove it?

Reply:
Can you try out a different regex and see if this works? See e.g. https://stackoverflow.com/q/2291428/194609

Reply:
I tried it like this:
```
@Path("/{profile}/{coordinatesArray : .+}")
...
@PathParam("coordinatesArray") String coordinates
```

I think this is what you meant? It does not work. The variable `coordinates` only contains the first coordinate up to the first `;`, everything after that seems to be read as matrix parameter.